### PR TITLE
feat: support signatureHelpProvider

### PR DESCRIPTION
**Pull request by bitaoz**
**Summary**
This adds the ability to prompt for a sas function and its parameters information after entering a function name. See #66 for more details.

**Testing**
Tested entering a normal sas function.
Tested entering a nested sas function within another sas function.
Tested entering a sas macro function.

### CODE REVIEW

File: client/test/extension.test.ts [hunk 1]
```diff
@@ -33,6 +33,18 @@ describe("lsp", () => {
     assert.ok(actualHover.contents[0]);
   });
 
+  it("provides signature help", async () => {
+    // Executing the command `vscode.executeCompletionItemProvider` to simulate triggering completion
```
Comment (scnwwu) [inline client/test/extension.test.ts:37]: please update comment, thanks

File: client/testFixture/SampleCode.sas [hunk 2]
```diff
@@ -52,3 +52,8 @@ proc univariate data=popstate freq plot normal;
 proc print data= univout;
 	title 'Output Dataset From PROC UNIVARIATE';
 	format popn50 pop_50 pop_95 pop_97_5 pop_100 best8.;
+proc iml;
```
Comment (scnwwu) [inline client/testFixture/SampleCode.sas:55]: This sample code is used in various places. I would suggest to not change it here. The specific test can edit active editor on the fly.

**Review by scnwwu — APPROVED** [review]

**Comment by Zhirong2022** [comment]
I did not go through all functions , however I got some issues in testing. I'd like to categorize them below and update it if having any new findings.

1. The parameter is not shown as the parameter name instead of the parameter values
open(), attrc(), attrn(), datdif(), btrim(), catq(), filename(), find(), findc(), findw(), fmtinfo(), getoption(), git_branch_merge(), open(), typeof(), week(), %sysevalf(), %sysmexecname()
2. It has no capability to change the subsequent parameters base on the specified one
cdf(), finance()
3. The parameters inconsistency between the prompt page and help doc
clibexist(), coalesce(), complev(), compress(), encodcompat(), euclid(), exist(), fopen(),fsep(), getsessopt(), geodist(), intfmt(), lexcombi(), rename(), run_macro(), savings(), second(), %kscan(), %qkscan(), %qscan(), %qsysfunc(), %sysfunc(), %validchs(), fdelete(), input(), x=dur(), gcd(), kurtosis(), quantile(),  tzoneu2s()
4. There is only one parameter on prompt page if having '|' in function help doc
dropnote(), %kcmpres(), %qkcmpres()
5. No function help 
substr(), %kleft(), %klength(), %kupcase(), %qkleft(), %qkupcase()

**Comment by Zhirong2022** [comment]
The function help prompt is not proper with typing. 
image

**Comment by Zhirong2022** [comment]
The function help is shown unexpectedly in certain case.
image

**Comment by Zhirong2022** [comment]
It has no function help prompt if the function positions at the left side of an expression.
image

**Pull request closed without merging**